### PR TITLE
Fix tensorflow 1.6.0 API change for Layer class

### LINKED ISSUE #1048: Perslay tests are failing with tensorflow ~~2.15.1~~ 2.16
While testing a new compilation environment with up-to-date third parties libraries, I faced errors with perslay tests:

```txt
      File "/workspace/gudhi/gudhi-devel/build/src/python/gudhi/tensorflow/perslay.py", line 132, in __init__
        super().__init__(dynamic=True, **kwargs)
      File "/miniconda3/envs/custom-cgal/lib/python3.10/site-packages/keras/src/layers/layer.py", line 263, in __init__
        raise ValueError(
    ValueError: Unrecognized keyword arguments passed to GaussianPerslayPhi: {'dynamic': True}
```

The perslay python tests is ok with tensorflow 2.11.0, but fails with tensorflow 2.15.1

### PR DESCRIPTION
**Pull request by VincentRouvreau**
Fix #1048

### CODE REVIEW

File: src/python/gudhi/tensorflow/_layer.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class TensorflowKerasLayer(tensorflow.keras.layers.Layer):
+    """
+    TensorFlow Keras layer
+    """
+    def __init__(self, **kwargs):
+        """
+        Constructor for the TensorflowKerasLayer class
+        """
+        # On tensorflow < 2.16.0 set dynamic argument to True
+        if parse_version(tensorflow.__version__) < parse_version('2.16.0'):
```
Comment (mglisse) [inline src/python/gudhi/tensorflow/_layer.py:14]: #1048 says that 2.15.1 rejects `dynamic=True`, I am confused...

Comment (VincentRouvreau) [reply src/python/gudhi/tensorflow/_layer.py:14]: I tried again with master branch of gudhi and python 3.11. The problem seems to occur on tensorflow >= 2.16.
After reading the [tensorflow 2.16.0 release note](https://github.com/tensorflow/tensorflow/releases/tag/v2.16.0-rc0), this change seems to come from `keras.layers.experimental.DynamicEmbedding` change.

Sorry about that, I was pretty sure it happens on tensorflow 2.15.1, but after several tests I can guarantee it happens on tensorflow 2.16.1 (no pip package for tensorflow 2.15.0)

File: src/python/gudhi/tensorflow/_layer.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import tensorflow
+# The sklearn parse_version function is taken from 'packaging' and this dependency is not mandatory
+from sklearn.utils.fixes import parse_version
```
Comment (mglisse) [inline src/python/gudhi/tensorflow/_layer.py:3]: I am not sure what the comment means.
Do we now need to install sklearn to be able to use the tensorflow stuff?
Would `tuple(map(int, tf.__version__.split('.')[:2])) < (2,16)` be good enough?

**Comment by VincentRouvreau** [comment]
Duplicate of #1063

**Pull request closed without merging**